Source/Dafny/Resolver.cs

```diff
@@ -3768,6 +3768,9 @@ public class ModuleBindings
         var lhsWithProxyArgs = Type.HeadWithProxyArgs(lhs);
         ConstrainSubtypeRelation(lhsWithProxyArgs, rhs, errMsg, false, allowDecisions);
         ConstrainAssignableTypeArgs(lhs, lhsWithProxyArgs.TypeArgs, lhs.TypeArgs, errMsg, out moreXConstraints);
+        if (Type.SameHead(lhs, rhs, true) && lhs.AsCollectionType == null) {
+          ConstrainAssignableTypeArgs(lhs, lhs.TypeArgs, rhs.TypeArgs, errMsg, out moreXConstraints);
```

Comment:
This will overwrite the `moreXConstraints` that was set in L3770. I think L3772 should probably pass a new local boolean variable and then update `moreXConstraint` to be the disjunction of `moreXConstraints` from L3700 and the new local variable.

Reply:
Done

Reply:
Fixed this